lib/ansible/module_utils/asa.py

```diff
@@ -42,33 +42,41 @@
     'authorize': dict(fallback=(env_fallback, ['ANSIBLE_NET_AUTHORIZE']), type='bool'),
     'auth_pass': dict(fallback=(env_fallback, ['ANSIBLE_NET_AUTH_PASS']), no_log=True),
     'timeout': dict(type='int'),
-    'provider': dict(type='dict'),
     'context': dict(),
     'passwords': dict()
 }
 
+asa_argument_spec = {
+    'provider': dict(type='dict', options=asa_provider_spec),
+}
+
+asa_top_spec = {
+    'host': dict(removed_in_version=2.3),
+    'port': dict(removed_in_version=2.3, type='int'),
+    'username': dict(removed_in_version=2.3),
+    'password': dict(removed_in_version=2.3, no_log=True),
+    'ssh_keyfile': dict(removed_in_version=2.3, type='path'),
+    'authorize': dict(type='bool'),
```

Comment:
Why is removed_in_version missing from this?

Reply:
In current implementation warning in not thrown if `authorize` is mentioned at top level. 
To keep the same behavior after refactor `removed_in_version` is not added for `authorize`

Reply:
asa_config doc [here](http://docs.ansible.com/ansible/latest/asa_config_module.html) confirms the same.